subsystem/ams_subscriptions_test.go

```diff
@@ -0,0 +1,69 @@
+package subsystem
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-openapi/strfmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/client"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+)
+
+var _ = Describe("test AMS subscriptions", func() {
+	ctx := context.Background()
+
+	BeforeEach(func() {
+		if !Options.EnableAuth {
+			Skip("auth is disabled")
+		}
+	})
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	Context("AMS subscription creation", func() {
+		It("happy flow", func() {
+			userClusterID, err := registerCluster(ctx, userBMClient, "user-cluster", fmt.Sprintf(psTemplate, FakePS))
+			Expect(err).ShouldNot(HaveOccurred())
+			cluster := getCommonCluster(ctx, userClusterID)
+			Expect(cluster.AmsSubscriptionID).Should(Equal(FakeSubscriptionID))
+		})
+		It("CreateSubscription failed", func() {
+			_, err := registerCluster(ctx, unallowedUserBMClient, "user-cluster", fmt.Sprintf(psTemplate, FakePS))
+			Expect(err).Should(HaveOccurred())
+		})
+	})
+
+	Context("AMS subscription update", func() {
+		It("happy flow", func() {
+			userClusterID, err := registerCluster(ctx, userBMClient, "user-cluster", fmt.Sprintf(psTemplate, FakePS))
+			Expect(err).ShouldNot(HaveOccurred())
+			registerHostsAndSetRoles(userClusterID, 3)
+			installCluster(userClusterID)
+			completeInstallation(ctx, userBMClient, userClusterID)
```

Comment:
this will require agent auth, see: https://github.com/openshift/assisted-service/blob/master/swagger.yaml#L1236-L1237

Reply:
done.